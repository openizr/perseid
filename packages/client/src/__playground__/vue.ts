/* c8 ignore start */

import {
  Router,
} from 'scripts/vue/index';
import {
  i18n,
  model,
  store,
  apiClient,
} from '__playground__/services';
import '__playground__/style.scss';
import { type Component, createApp } from 'vue';

function main(): void {
  // Creating Vue root...
  const app = createApp(Router as unknown as Component, {
    container: document.querySelector('#root'),
    services: {
      i18n,
      model,
      store,
      apiClient,
    },
  });
  (app as { mount: (root: string) => void; }).mount('#root');
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}
