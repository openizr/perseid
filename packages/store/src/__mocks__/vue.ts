/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

export const ref = vi.fn(() => ({ value: 'test' }));
export const onMounted = vi.fn((callback: () => unknown) => callback());
export const onUnmounted = vi.fn((callback: () => unknown) => callback());
