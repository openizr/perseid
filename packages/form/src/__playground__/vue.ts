/* c8 ignore start */

import Form from 'scripts/vue/Form.vue';
import Engine from 'scripts/core/Engine';
import Step from 'scripts/vue/DefaultStep.vue';
import { type Component, createApp } from 'vue';

let app: unknown;
const { log } = console;

function main(): void {
  app = createApp(Form as unknown as Component, {
    stepComponent: Step,
    configuration: {
      root: 'root',
      fields: {
        test: { type: 'string', required: true },
      },
      steps: {
        root: {
          fields: ['test'],
        },
      },
      plugins: [
        (engine: Engine): void => {
          engine.on('userAction', (data, next) => {
            log(data);
            return next(data);
          });
        },
      ],
    },
  });
  (app as { mount: (root: string) => void; }).mount('#root');
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting Vue components tree.
window.addEventListener('beforeunload', () => {
  (app as { unmount: () => void; }).unmount();
});
