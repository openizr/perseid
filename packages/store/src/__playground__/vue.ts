/* c8 ignore start */

import App from '__playground__/App.vue';
import { Component, createApp } from 'vue';

let app: unknown;

function main(): void {
  app = createApp(App as unknown as Component);
  (app as { mount: (root: string) => void; }).mount('#root');
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting Vue components tree.
window.addEventListener('beforeunload', () => {
  (app as { unmount: () => void; }).unmount();
});
