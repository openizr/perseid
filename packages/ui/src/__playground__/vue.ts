/* c8 ignore start */

import { createApp, type App } from 'vue';
import AppRouter from '__playground__/pages/AppRouter.vue';

let app: App;

function main(): void {
  app = createApp(AppRouter as unknown as App);
  app.mount('#root');
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting Vue components tree.
window.addEventListener('beforeunload', () => {
  app.unmount();
});
