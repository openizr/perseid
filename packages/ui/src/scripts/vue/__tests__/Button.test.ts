/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @vitest-environment jsdom
 */

import UIButton from 'scripts/vue/UIButton.vue';
import { render, fireEvent } from '@testing-library/vue';

describe('vue/UIButton', () => {
  vi.mock('scripts/core/index');
  vi.mock('scripts/vue/UIIcon.vue');

  beforeEach(() => {
    vi.clearAllMocks();
  });

  test('renders correctly - basic', () => {
    const { container } = render(UIButton, { props: { label: 'Test', modifiers: 'large' } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - with id', () => {
    const { container } = render(UIButton, { props: { id: 'test', label: 'Test', modifiers: 'large' } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - no label', () => {
    const { container } = render(UIButton, { props: {} });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - label with icon', () => {
    const { container } = render(UIButton, { props: { label: 'Test', icon: 'star' } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - type submit', () => {
    const { container } = render(UIButton, { props: { label: 'Test', type: 'submit' } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - icon only', () => {
    const { container } = render(UIButton, { props: { icon: 'star' } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - right icon', () => {
    const { container } = render(UIButton, { props: { icon: 'star', iconPosition: 'right' } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - disabled', () => {
    const { container } = render(UIButton, { props: { disabled: true } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - with listener', async () => {
    const onClick = vi.fn();
    const onFocus = vi.fn();
    const { container } = render(UIButton, {
      props: {
        onClick,
        onFocus,
      },
    });
    const button = container.getElementsByTagName('button')[0];
    await fireEvent.click(button);
    await fireEvent.focus(button);
    expect(container.firstChild).toMatchSnapshot();
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith(expect.any(Object));
    expect(onFocus).toHaveBeenCalledTimes(1);
    expect(onFocus).toHaveBeenCalledWith(expect.any(Object));
  });
});
