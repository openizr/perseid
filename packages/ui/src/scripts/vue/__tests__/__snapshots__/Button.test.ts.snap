// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vue/UIButton > renders correctly - basic 1`] = `
<button
  class="ui-button ui-button--large"
  tabindex="0"
  type="button"
>
  <!--v-if-->
  <span
    class="ui-button__label"
  >
    Test
  </span>
  <!--v-if-->
</button>
`;

exports[`vue/UIButton > renders correctly - disabled 1`] = `
<button
  class="ui-button ui-button--disabled"
  tabindex="-1"
  type="button"
>
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</button>
`;

exports[`vue/UIButton > renders correctly - icon only 1`] = `
<button
  class="ui-button ui-button--icon"
  tabindex="0"
  type="button"
>
  <i
    class="ui-icon ui-icon--star"
  />
  <!--v-if-->
  <!--v-if-->
</button>
`;

exports[`vue/UIButton > renders correctly - label with icon 1`] = `
<button
  class="ui-button"
  tabindex="0"
  type="button"
>
  <i
    class="ui-icon ui-icon--star"
  />
  <span
    class="ui-button__label"
  >
    Test
  </span>
  <!--v-if-->
</button>
`;

exports[`vue/UIButton > renders correctly - no label 1`] = `
<button
  class="ui-button"
  tabindex="0"
  type="button"
>
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</button>
`;

exports[`vue/UIButton > renders correctly - right icon 1`] = `
<button
  class="ui-button ui-button--icon"
  tabindex="0"
  type="button"
>
  <!--v-if-->
  <!--v-if-->
  <i
    class="ui-icon ui-icon--star"
  />
</button>
`;

exports[`vue/UIButton > renders correctly - type submit 1`] = `
<button
  class="ui-button"
  tabindex="0"
  type="submit"
>
  <!--v-if-->
  <span
    class="ui-button__label"
  >
    Test
  </span>
  <!--v-if-->
</button>
`;

exports[`vue/UIButton > renders correctly - with id 1`] = `
<button
  class="ui-button ui-button--large"
  id="test"
  tabindex="0"
  type="button"
>
  <!--v-if-->
  <span
    class="ui-button__label"
  >
    Test
  </span>
  <!--v-if-->
</button>
`;

exports[`vue/UIButton > renders correctly - with listener 1`] = `
<button
  class="ui-button"
  tabindex="0"
  type="button"
>
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</button>
`;
